File: flutter_app/lib/features/outlets/models/outlet.dart
import 'package:flutter/foundation.dart';

@immutable
class Outlet {
  const Outlet({
    required this.id,
    required this.name,
    this.address,
    this.phone,
    required this.isActive,
    required this.createdAt,
  });

  final String id;
  final String name;
  final String? address;
  final String? phone;
  final bool isActive;
  final DateTime createdAt;

  factory Outlet.fromJson(Map<String, dynamic> json) => Outlet(
        id: json['id'] as String,
        name: json['name'] as String,
        address: json['address'] as String?,
        phone: json['phone'] as String?,
        isActive: (json['is_active'] as bool?) ?? true,
        createdAt: DateTime.parse(json['created_at'] as String),
      );
}

File: backend/internal/models/outlet.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Outlet struct {
	ID        uuid.UUID `json:"id"`
	Name      string    `json:"name"`
	Address   *string   `json:"address,omitempty"`
	Phone     *string   `json:"phone,omitempty"`
	IsActive  bool      `json:"is_active"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type CreateOutletRequest struct {
	Name     string  `json:"name"`
	Address  *string `json:"address,omitempty"`
	Phone    *string `json:"phone,omitempty"`
	IsActive *bool   `json:"is_active,omitempty"`
}

type UpdateOutletRequest struct {
	Name     *string `json:"name,omitempty"`
	Address  *string `json:"address,omitempty"`
	Phone    *string `json:"phone,omitempty"`
	IsActive *bool   `json:"is_active,omitempty"`
}
